pkg/controller/node/nodecontroller.go

```diff
@@ -496,6 +488,43 @@ func (nc *NodeController) monitorNodeStatus() error {
 	return nil
 }
 
+var nodeExistsInCloudProvider = func(nc *NodeController, nodeName string) (bool, error) {
+	instances, ok := nc.cloud.Instances()
+	if !ok {
+		return false, fmt.Errorf("%v", ErrCloudInstance)
+	}
+	if _, err := instances.ExternalID(nodeName); err != nil {
+		if err == cloudprovider.InstanceNotFound {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// forcefullyDeleteNode immediately deletes all pods on the node, and then
+// deletes the node itself.
+func (nc *NodeController) forcefullyDeleteNode(nodeName string) error {
+	selector := fields.OneTermEqualSelector(client.PodHost, nodeName)
+	options := api.ListOptions{FieldSelector: selector}
+	pods, err := nc.kubeClient.Core().Pods(api.NamespaceAll).List(options)
+	if err != nil {
+		return fmt.Errorf("unable to list pods on node %q: %v", nodeName, err)
+	}
+	for _, pod := range pods.Items {
```

Comment:
If kubelet is semi-functioning, it may try to recreate the mirror pods. You'll be left with running pods on a node that doesn't exist anymore. Since this PR deletes pods/nodes only when the vm has disappeared, it should be fine.